test/coverage/datamodel/UuidValidatorTest.cpp

```diff
@@ -23,16 +26,216 @@ class TestFixture : public testing::Test {
+	EXPECT_FALSE(valid);
+	EXPECT_EQ(reason.value(),
+	          uprotocol::datamodel::validator::uuid::Reason::EXPIRED);
+}
+
+// Test non-expired UUID
+TEST_F(TestUuidValidator, NonExpiredUuid) {
+	// Creating a UUID with a recent timestamp
+	uint64_t recent_timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count() -
+	    30000;  // 30 seconds in the past
+	uint64_t msb = (8ULL << 12) | (recent_timestamp << 16) |
+	               (0x123ULL);  // Version 8, recent timestamp
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isExpired(
+	    uuid, std::chrono::milliseconds(60000));  // 60 seconds TTL
+	EXPECT_TRUE(valid);
+	EXPECT_FALSE(reason.has_value());
+}
+
+// Test invalid counter value
+TEST_F(TestUuidValidator, InvalidCounterValue) {
+	// Creating a UUID with counter exceeding 4095
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | 0x1FFF;  // Version 8, counter = 8191
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isUuid(uuid);
+	EXPECT_FALSE(valid);
+}
+
+// Test retrieving version
+TEST_F(TestUuidValidator, RetrieveVersion) {
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_EQ(uprotocol::datamodel::validator::uuid::getVersion(uuid), 8);
+}
+
+// Test retrieving variant
+TEST_F(TestUuidValidator, RetrieveVariant) {
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_EQ(uprotocol::datamodel::validator::uuid::getVariant(uuid), 2);
+}
+
+TEST_F(TestUuidValidator, RetrieveTimestamp) {
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | (0x123ULL);  // Version 8, timestamp
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto uuid_time = uprotocol::datamodel::validator::uuid::getTime(uuid);
+	auto uuid_timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+	                          uuid_time.time_since_epoch())
+	                          .count();
+	EXPECT_EQ(uuid_timestamp, timestamp);
```

Comment:
See earlier comments about doing time arithmetic and comparisons using chrono objects